test/migration/verify/verify_migration_test.go

```diff
@@ -0,0 +1,161 @@
+package verify
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/states"
+	"github.com/codeready-toolchain/toolchain-e2e/test/migration"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/cleanup"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/md5"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/wait"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestAfterMigration(t *testing.T) {
+	// given
+	awaitilities := WaitForDeployments(t)
+
+	// get Signups for the users provisioned in the setup part
+	provisionedSignup := getSignupFromMUR(t, awaitilities.Host(), migration.ProvisionedUser)
+	secondMemberProvisionedSignup := getSignupFromMUR(t, awaitilities.Host(), migration.SecondMemberProvisionedUser)
+	appstudioProvisionedSignup := getSignupFromMUR(t, awaitilities.Host(), migration.AppStudioProvisionedUser)
+	deactivatedSignup := listAndGetSignupWithState(t, awaitilities.Host(), toolchainv1alpha1.UserSignupStateLabelValueDeactivated)
+	bannedSignup := listAndGetSignupWithState(t, awaitilities.Host(), toolchainv1alpha1.UserSignupStateLabelValueBanned)
```

Comment:
No, we cannot do that, because:
* we don't know the names of the UserSignups
* we know only the names of the MURs, but there is one account that is deactivated and one that is banned, so we cannot use simple get command - we have to list them using the `state` label
* the functions are executed in parallel and some of them also bans or deactivates the UserSignups to test that everything works fine, so the list could return more UserSignups than only one

Does it makes sense?

Reply:
Sorry, I'm still a bit confused as to why a function like this wouldn't work:
```
func verifyProvisionedSignup(t *testing.T, awaitilities wait.Awaitilities) {
	signup := getSignupFromMUR(t, awaitilities.Host(), migration.ProvisionedUser)
	cleanup.AddCleanTasks(t, awaitilities.Host().Client, signup)
	VerifyResourcesProvisionedForSignup(t, awaitilities, signup, "base")
	DeactivateAndCheckUser(t, awaitilities, signup)
	ReactivateAndCheckUser(t, awaitilities, signup)
}
```

It would still get the UserSignup using the MUR name.

For the deactivated/banned ones we could also do it within the verify function?
eg.
```
func verifyDeactivatedSignup(t *testing.T, awaitilities wait.Awaitilities) {
	signup := listAndGetSignupWithState(t, awaitilities.Host(), toolchainv1alpha1.UserSignupStateLabelValueDeactivated)
	cleanup.AddCleanTasks(t, awaitilities.Host().Client, signup)

	_, err := awaitilities.Host().WaitForUserSignup(signup.Name,
		wait.UntilUserSignupContainsConditions(ConditionSet(Default(), DeactivatedWithoutPreDeactivation())...),
		wait.UntilUserSignupHasStateLabel(toolchainv1alpha1.UserSignupStateLabelValueDeactivated))
	require.NoError(t, err)
	require.True(t, states.Deactivated(signup), "usersignup should be deactivated")

	err = awaitilities.Host().WaitUntilMasterUserRecordDeleted(migration.DeactivatedUser)
	require.NoError(t, err)

	ReactivateAndCheckUser(t, awaitilities, signup)
}
```

But what I just realized is that with the current way we're following the given, when, then testing pattern so I guess it makes sense to get the usersignups as part of the 'given' phase of the test. So you can disregard my comment.

Reply:
Actually, we could still follow the given, when, then pattern if we do:
```
func verifyProvisionedSignup(t *testing.T, awaitilities wait.Awaitilities) {
	//given
	signup := getSignupFromMUR(t, awaitilities.Host(), migration.ProvisionedUser)

	// when & then
	cleanup.AddCleanTasks(t, awaitilities.Host().Client, signup)
	VerifyResourcesProvisionedForSignup(t, awaitilities, signup, "base")
	DeactivateAndCheckUser(t, awaitilities, signup)
	ReactivateAndCheckUser(t, awaitilities, signup)
}
```

Reply:
so let's take the two functions that you mentioned as an example:
* `verifyProvisionedSignup`
* `verifyDeactivatedSignup`

these functions are executed in parallel. 

Now let's take a look at how the `listAndGetSignupWithState` function (which would be used in `verifyDeactivatedSignup`):
https://github.com/codeready-toolchain/toolchain-e2e/blob/8a83801fe2f3f14f29fd33ed199e6b6335ed2fca/test/migration/verify/verify_migration_test.go#L154-L161
it lists all `UserSignups` with the label `state=deactivated` and expects that there is only one `UserSignup` found.

However, the `verifyProvisionedSignup` - after checking that the `ProvisionedUser` is provisioned - also deactivates the `UserSignup`. 
If the deactivation happens before the `verifyDeactivatedSignup` calls `listAndGetSignupWithState`, then it could get two deactivated `UserSignups` instead of one. 

I know that this is really a corner case that should never happen, but still - I want to eliminate any flakyness.

Reply:
Ahh, ok now I see what you mean. Ok, that makes sense and I agree it's better to keep it there. We could probably add a little more to the comment to point out why it's better to have it there.